KUBERNETES.md

```diff
@@ -0,0 +1,231 @@
+certificate for TLS, it will result in a browser warning.
+
+### Operations Workflow
+
+> Note that this is an alternative to the Development workflow documented above.
+
+This workflow matches more closely how an application would be deployed in a production environment. In this method,
+the images are built by SBT. The [reactive-cli](https://github.com/lightbend/reactive-cli) is then
+used to generate YAML for deployment to the Kubernetes cluster. The steps below focus on Minikube, but could be used on
+a real Kubernetes cluster given access to a Docker registry and configured `kubectl`.
+
+> Want a fresh Minikube? Run `minikube delete` before the steps below.
+
+##### Start Minikube
+
+```bash
+minikube addons enable ingress
+minikube start --memory 6000
+```
+
+##### Setup Docker engine context to point to Minikube
+
+```bash
+eval $(minikube docker-env)
+```
+
+##### Install Reactive Sandbox (optional; skip if you have your own Cassandra, Kafka, and Elasticsearch clusters) 
+
+The `reactive-sandbox` includes development-grade (i.e. it will lose your data) installations of Cassandra, 
+Elasticsearch, Kafka, and ZooKeeper. It's packaged as a Helm chart for easy installation into your Kubernetes cluster.
+
+```bash
+helm init
+helm repo add lightbend-helm-charts https://lightbend.github.io/helm-charts
+helm repo update
+```
+
+Verify that Helm is available (this takes a minute or two):
+
+```bash
+kubectl --namespace kube-system get deploy/tiller-deploy
+```
+
+```
+NAME            DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+tiller-deploy   1         1         1            1           3m
+```
+
+Install the sandbox.
+
+```bash
+helm install lightbend-helm-charts/reactive-sandbox --name reactive-sandbox
+```
+
+Verify that it is available (this takes a minute or two):
+
+>  The `-w` flag will watch for changes. Use `CTRL-c` to exit.
+
+```bash
+kubectl get -w deploy/reactive-sandbox
+```
+
+```
+NAME               DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+reactive-sandbox   1         1         1            1           1m
+```
+
+##### Build Project
+
+To build all of the subprojects, run the following command:
+
+```bash
+sbt clean docker:publishLocal
+```
+
+To build a single subproject, prefix the `docker:publishLocal` task with the project's name and a forward slash. For instance,
+the following will only build `itemImpl`:
+
+``bash
+sbt itemImpl/docker:publishLocal
+``
+
+##### View Images
+
+```bash
+docker images
+```
+
+##### Deploy Projects
+
+Finally, you're ready to deploy the services. Be sure to adjust the secret variables and cassanda service address as necessary.
+
+```bash
+#
+# NOTE: You must change the secret values below or the applications will crash.
+#
+# These values are used for Play's application secret. It is important that they are set to a secret value.
+# More information: https://www.playframework.com/documentation/latest/ApplicationSecret
```

Comment:
A random (!) thought I had was: Maybe Play could specialcase `random`? Similar to `changeme`, but instead of failing, replacing it with a random UUID or something? Or alternatively, our guides could pass some random data in as part of the bash scripts.

Leaving it as-is for now, as it's probably not really worth the effort to change.

Reply:
The thing is, you really need to make sure that this stays consistent between executions, or else it won't be able to decrypt any cookies.

This is really only an issue for Play webapps, and Lagom services ideally shouldn't need to set a secret at all unless you invoke functionality that requires it (rare for Lagom services). See https://github.com/playframework/playframework/issues/8075

In a perfect world, we'd demonstrate how to provision this using k8s secrets (or something more secure).